shared/AppInsightsCore/Tests/Unit/src/StatsBeat.Tests.ts

```diff
@@ -0,0 +1,260 @@
+                
+                // Check that exception metrics are tracked
+                let foundExceptionMetric = false;
+                for (let i = 0; i < this._trackSpy.callCount; i++) {
+                    const call = this._trackSpy.getCall(i);
+                    const item: ITelemetryItem = call.args[0];
+                    if (item.baseData && 
+                        item.baseData.properties && 
+                        item.baseData.properties.exceptionType === "NetworkError") {
+                        foundExceptionMetric = true;
+                        break;
+                    }
+                }
+                
+                Assert.ok(foundExceptionMetric, "Exception metrics should be tracked");
+            }
+        });
+
+        this.testCase({
+            name: "StatsBeat: does not send metrics for different endpoints",
+            test: () => {
+                // Initialize StatsBeat for a specific endpoint
+                this._statsbeat.initialize(this._core, {
+                    ikey: "Test-iKey",
+                    endpoint: "https://example.endpoint.com",
+                    version: "1.0.0"
+                });
+                
+                // Create mock payload data
+                const payloadData = {
+                    urlString: "https://example.endpoint.com",
+                    data: "testData",
+                    headers: {},
+                    timeout: 0,
+                    disableXhrSync: false,
+                    statsBeatData: {
+                        startTime: Date.now()
+                    }
+                } as IPayloadData;
+                
+                // Set up spies to check internal calls
+                const countSpy = this.sandbox.spy(this._statsbeat, "count");
+                
+                // Count metrics for a different endpoint
+                this._statsbeat.count(200, payloadData, "https://different.endpoint.com");
+                
+                // The count method was called, but it should return early
+                Assert.equal(1, countSpy.callCount, "count method should be called");
+                Assert.equal(0, this._trackSpy.callCount, "track should not be called for different endpoint");
+            }
+        });
+
+        this.testCase({
+            name: "StatsBeat: test dynamic configuration changes",
+            useFakeTimers: true,
+            test: () => {
+                // Setup core with statsbeat enabled
+                this._core.initialize(this._config, [new ChannelPlugin()]);
+                
+                // Verify that statsbeat is created
+                const statsbeat = this._core.getStatsBeat();
+                Assert.ok(statsbeat, "Statsbeat should be created");
+                
+                // Update configuration to disable statsbeat
+if (!this._core.config._sdk) {
+                this._core.config._sdk = {};
```

Comment:
Does this really "remove" the `_sdk` object, if so we should update the default setting of the `_sdk` configuration to be a reference so that any defaults and any merged config (like fetching from the CDN) should "merge" not replace all of the properties of this object.

Reply:
And then explicitly set the value as required.

As by "removing" the reference this will actually break the dynamic configuration as any subsequent changes after it's removed will cause the automatic tracking to also get removed, so any "changes" after removal will actually not be tracked.

Reply:
Once you make the change I've just added to the core, this code as is "should" work correctly, although, technically the `_sdk` object should now never not exist